tap_mambu/helpers/__init__.py

```diff
@@ -127,3 +129,29 @@ def get_selected_streams(catalog):
         if root_metadata and root_metadata.get('selected') is True:
             selected_streams.add(stream.tap_stream_id)
     return list(selected_streams)
+
+
+# Currently syncing sets the stream currently being delivered in the state.
+# If the integration is interrupted, this state property is used to identify
+#  the starting point to continue from.
+# Reference: https://github.com/singer-io/singer-python/blob/master/singer/bookmarks.py#L41-L46
+def update_currently_syncing(state, stream_name):
+    if (stream_name is None) and ('currently_syncing' in state):
+        del state['currently_syncing']
+    else:
+        singer.set_currently_syncing(state, stream_name)
+    singer.write_state(state)
+
+
+# Review last_stream (last currently syncing stream), if any,
+#  and continue where it left off in the selected streams.
+# Or begin from the beginning, if no last_stream, and sync
+#  all selected steams.
+# Returns should_sync_stream (true/false) and last_stream.
+def should_sync_stream(selected_streams, last_stream, stream_name):
+    if last_stream == stream_name or last_stream is None:
+        if last_stream is not None:
+            last_stream = None
+        if stream_name in selected_streams:
+            return True, last_stream
+    return False, last_stream
```

Comment:
The `get_selected_streams` in the Catalog module will do a similar task to this, except it doesn't skip preceding streams if they're selected. Instead, it will shuffle the list of selected streams based on the currently syncing stream before returning the ordered set. https://github.com/singer-io/singer-python/blob/master/singer/catalog.py#L151

We've found that in practice, it's better to attempt all streams when resuming rather than wait for the next scheduled extraction, since sometimes folks have their frequency set at many hours or a whole day.

Reply:
We'll consider replacing this code with the included function